app/models/submission.rb

```diff
@@ -53,6 +54,8 @@ class Submission < ApplicationRecord
   validate :can_be_ineligible_for_evaluation,
            if: -> { judging_status_change == %w[selected not_selected] }
 
+  after_save :update_submission_evaluation_status
```

Comment:
I think we want to avoid the additional queries as a callback every time a submission is saved, it is somewhat expensive to query all these join records. Instead it can be a normal instance method that is called whenever there are specific events or updates in the system that could trigger the `evaluation_status` change. then it can be scoped more narrowly to that particular change.
```suggestion
```
please reach out if I'm wrong and removing this breaks something, but reviewing my state transition diagram I don't see anything that would update the status directly from a change to the submission model.

Reply:
one thing we **could** do would be to update the existing submissions in the DB via a rake task in order to backfill the correct statuses. however, I can accomplish this via a small console script and run it on the command line after deployment to staging, so it's not needed here.